src/engine/enginepregain.cpp

```diff
@@ -51,36 +53,46 @@ EnginePregain::EnginePregain(QString group)
 EnginePregain::~EnginePregain()
 {
     delete m_pPotmeterPregain;
-    delete m_pControlReplayGain;
+    delete m_pCOReplayGain;
     delete m_pTotalGain;
 
     delete s_pEnableReplayGain;
     s_pEnableReplayGain = NULL;
     delete s_pReplayGainBoost;
     s_pReplayGainBoost = NULL;
+    delete s_pDefaultBoost;
+    s_pDefaultBoost = NULL;
 }
 
 void EnginePregain::process(CSAMPLE* pInOut, const int iBufferSize) {
-
-    float fEnableReplayGain = s_pEnableReplayGain->get();
-    const float fReplayGainBoost = s_pReplayGainBoost->get();
-    float fGain = m_pPotmeterPregain->get();
-    float fReplayGain = m_pControlReplayGain->get();
-    float fReplayGainCorrection = 1;
-    float fPassing = m_pPassthroughEnabled->get();
-
-    // Override replaygain value if passing through
-    if (fPassing > 0) {
-        fReplayGain = 1.0;
-    } else if (fReplayGain * fEnableReplayGain != 0) {
-        // Here is the point, when ReplayGain Analyser takes its action, suggested gain changes from 0 to a nonzero value
+    const float fReplayGain = m_pCOReplayGain->get();
+    float fReplayGainCorrection;
+    if (m_pPassthroughEnabled->toBool()) {
+        // Override replaygain value if passing through
+        // TODO(XXX): consider a good default.
+        // Do we expect an replaygain leveled input or
+        // Normalized to 1 input?
+        fReplayGainCorrection = 1; // We expect a replaygain leveled input
+    } else if (!s_pEnableReplayGain->toBool() || fReplayGain == 0) {
```

Comment:
I like to have a feature to tweak the center position of the deck gain knob. 

If we apply the defaultBoost in any case, it comes for free, without the need for other additional controls. 
Currently we do not need it that hard, but once we have the floating point API, it is a common case that the lossy coded tracks have samples above 1, which requires to tweak the gain by default. 
I hope you understand that having it allways on is just an additional degree of freedom, to issue such situations and I cannot think of a situation where this can be annoying for specific Mixing style.
This "feature" will be lost if we reset the boost to 0 dB if replayGain correction is disabled.  

What could be an alternative solution or improvement to get out of the deadlock in our discussion? 

Would it help to move the defaultBoost slider to 0 when unchecking the ReplayGain enable box?